packages/devtools/src/microsoft/teams/devtools/devtools_plugin.py

```diff
@@ -49,7 +49,7 @@ class DevToolsPlugin(Sender):
     logger: Annotated[Logger, LoggerDependencyOptions()]
     id: Annotated[Optional[TokenProtocol], DependencyMetadata(optional=True)]
     name: Annotated[Optional[TokenProtocol], DependencyMetadata(optional=True)]
```

Comment:
One potential nicer thing could be this:
1. When registering a particular dep, we can register with "is_optional".
2. When we're parsing this, we can check if it's Annotated[Optional[...]. If it's not, we can warn that the plugin is treating an optional plugin as non-optional.

Reply:
pushing ideation/discussion to next PR to unblock Mehak!